### DISCUSSION TITLE
Timeout Error when connecting to camera in different network

### DISCUSSION THREAD

**Discussion by wiliamfjr** [thirtythreeforty/neolink#106] [General]
Hello...

I encountered a problem when I tried to connect to my reolink E1 from different network
`Error streaming from camera, will retry in 1s: Timeout`
When I tried it in the same network as the camera, it worked perfectly.

Thanks for the help!

**Comment by twistedddx** [comment]
williamfjr, do you know how to use wireshark?
Can you tell if the camera is trying to communicate via TCP or UDP to Neolink?

I wonder if cams switch to BC UDP when connection comes from a different network?
If so this would require Neolink to implement BC UDP which is a feature request that several other issues are also depending on.

**Comment by QuantumEntangledAndy** [comment]
Could you please explain what you mean by same and different network? Crossing networks usually needs a little extra setup depending on what your trying to achieve.

**Comment by wiliamfjr** [comment]
Hi, thanks for replying, what I tried to do is connect my camera to Office 1 network and stream it in Office 2 network. Right now I'm testing it with my Office and mobile network, and while the video from Reolink Apps seems to run correctly, I get Timeout Error when trying it with neolink.
I'm not sure about the protocol and will check it tomorrow

**Comment by QuantumEntangledAndy** [comment]
Your office may not be allowing network traffic to cross the two networks. Can you ping the camera from office2? Can you ping the machine with neolink from office2.

**Comment by QuantumEntangledAndy** [comment]
Reolink apps use a process called NAT hole punching to get around these network issues where they route the connection through their servers

**Comment by Lanwalker** [comment]
Sounds like they are using Amazon's cloud to route thru...

The ' new ' Reolink client has issues and I went back to an older version. Battery powered units do not expose the ' UID ' under the CAMERA ... INFO like the E1 units do and the networking setup is basically gone from the newer version.

They also removed the magnify function and put in a ' CLIP ' and trim that really sucks to me.

I told Reolink support about the issues... and told them I was going back to the older version until they got things fixed. They sent me a link to a different version of the ' new ' client that still sucked and the Chinese version was included and I had to use task manager to kill that thing... it would not respond or quit / exit .

I told them about those issues too. We'll see what they come up with next. Both of their new ' old ' versions wanted to poke holes thru my firewalls... not real happy about that either.

**Comment by thirtythreeforty** [comment]
Agree with @Lanwalker and @QuantumEntangledAndy that it's bouncing through the Reolink cloud for the official app. Unfortunately this is one of the privileges of network administration - you'll have to figure out why you can't route.  If you're using TCP, everything should work.

Happy to help continue troubleshooting; I'm going to convert this to a Discussion.
